### PR TITLE
Added the ability to receive Me Events.

### PR DESCRIPTION
**Pull request by Pocketpac**
Thought you might find this useful. Figured if I can help with this java client since I am deeply invested in it, i might as well.

### CODE REVIEW

**Comment by Pocketpac** [comment]
Actually I noticed just some junk that I wanna clean out. Just a few imports I dont want in there because its pointless, will re-pull request this.

**Comment by connor4312** [comment]
You can leave the PR open. Just continue committing to the repo, and your commits will be amended into the request.

**Comment by Pocketpac** [comment]
Oh, interesting. I am somewhat comfortable with the coding side of things, though I am no where near Taylor and your level. But this whole fork/pull thing is new to me. ;-)

**Comment by connor4312** [comment]
No problem, there's always stuff for everyone to learn ;)

*throws ball over to @ttaylorr*

**Comment by ttaylorr** [comment]
:sparkles: Great, the code that you added here should do the trick.

Before I merge, I noticed a lot of formatting changes that yield the same code (I assume your editor's auto-format did this).  [One example is here](https://github.com/MCProHosting/beam-client-java/pull/22/files#diff-7dfa617ced618734f68ff453d189690eL23).

If you could clean those up, I'd be more than happy to merge this!

And once again, thank you _so_ much for submitting this patch.

File: src/main/java/pro/beam/api/resource/chat/AbstractChatEvent.java [hunk 1]
```diff
@@ -11,14 +17,25 @@ public AbstractChatEvent() {
...
+        @SerializedName("PollEnd")
+        POLL_END(PollEndEvent.class),
+        @SerializedName("Stats")
+        STATS(StatusEvent.class),
+        @SerializedName("UserJoin")
+        USER_JOIN(UserJoinEvent.class),
+        @SerializedName("UserLeave")
+        USER_LEAVE(UserLeaveEvent.class),
+        @SerializedName("MeEvent")
+        ME_EVENT(UserLeaveEvent.class);
```
Comment (ttaylorr) [inline src/main/java/pro/beam/api/resource/chat/AbstractChatEvent.java]: I think the `ME_EVENT` enum should serialize to a `MeEvent.class`.  i.e.,

```java
public static enum EventType {
  @SerializedName("MeEvent")
  ME_EVENT(MeEvent.class);
}
```

**Comment by ttaylorr** [comment]
Basically, I think we just need the diff from 5dc603f5407828bacf6dc88310f3aef4fb0ed529.  There are a few un-necessary changes in that diff, but I can clean those up later.  If you want to [rebase](http://git-scm.com/book/en/v2/Git-Branching-Rebasing) this branch down to just that commit, that should do the trick.

**Comment by Pocketpac** [comment]
I had it setup to use the MeEvent, but apparently somehow I reverted it back to before I made that change. And by cleaning up the formatting I assume you are referring to making the Serializedname's all single lines like you had it? Yes I have NetBeans auto-format makes it easier on my eyes

**Comment by ttaylorr** [comment]
Yes, but there are other parts of the code that were touched by NetBeans' formatting tool.  Some of the imports were changed in order, and other misc things.

**Comment by Pocketpac** [comment]
There, I fix it so the MeEvent serializes properly again. Guessing I did an undo at some point and didn't realize it.

**Comment by Pocketpac** [comment]
You know what, I am going to flush and rebuild this from scratch. I'm closing this pull request and will re-pull after a bit.

**Pull request closed without merging**